modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/FrontendDataSet.js

```diff
@@ -273,6 +277,33 @@ const FrontendDataSet = ({
 		}
 	}
 
+	const pageSelectedItemsValue = selectedItemsValue.filter((id) =>
+		items.some((item) => item.id === id)
+	);
```

Comment:
This is a quadratic complexity loop. It's probably not a huge issue only because we know items count should always be low, but if some customers increased number of items per page, this could explode. Is there a smarter way to do this?

This is also done on each render :grimacing: Besides the quadratic complexity thing, just the fact that we are making state changes on each render makes this look extremely fragile.

Have you considered the solution where we don't need to keep track of  any of this, that I mentioned in https://github.com/liferay-frontend/liferay-portal/pull/4821#discussion_r2025128569

Reply:
Oh, I realized the quadratic complexity thing was already in master, you just changed where it is invoked :disappointed:  Still, now it would be much worse since it it is in root of the main component, so it would re-calculate every single time on render, and FDS re-renders a lot.